Mlem/Views/Shared/Cached Image.swift

```diff
@@ -13,11 +14,14 @@ import QuickLook
 import SwiftUI
 
 struct CachedImage: View {
+    @Dependency(\.notifier) var notifier
     let url: URL?
     let shouldExpand: Bool
+    let hasContextMenu: Bool
     
     // state vars to track the current image size and whether that size needs to be recomputed when the image actually loads. Combined with the image size cache, this produces good scrolling behavior except in the case where we scroll past an image and it derenders before it ever gets a chance to load, in which case that image will cause a slight hiccup on the way back up. That's kind of an unsolvable problem, since we can't know the size before we load the image at all, but that's fine because it shouldn't really happen during normal use. If we really want to guarantee smooth feed scrolling we can squish any image with no cached size into a square, but that feels like squishing a lot of images for the sake of a fringe case.
     @State var size: CGSize
+    let fixedSize: CGSize?
```

Comment:
Request some documentation for `let fixedSize`, seems to be used to set frame but also absence of it means previews aren't available?

Reply:
When fixedSize is used, we need to explicitly set the preview so that the image appears at the correct aspect ratio in the context menu. This happens in the thumbnails in compact mode.

Showing the preview would also theoretically work in all other cases too, but unfortunately a small transition bug occurs if we state the preview explicitly for images in large posts. To avoid this, I don't state the preview, and SwiftUI uses the view itself as the preview. This has no visual difference to setting a preview explicitly, but fixes the aforementioned transition bug